src/main/java/net/earthcomputer/clientcommands/command/ItemGroupCommand.java

```diff
@@ -0,0 +1,361 @@
+        return 0;
+    }
+
+    private static int changeIcon(ServerCommandSource source, String name, ItemStack icon) throws CommandSyntaxException {
+        if (!groups.containsKey(name)) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        Group group = groups.get(name);
+        ItemGroup itemGroup = group.getItemGroup();
+        ListTag items = group.getItems();
+        ItemStack old = itemGroup.getIcon();
+        itemGroup = FabricItemGroupBuilder.create(
+                new Identifier("clientcommands", name))
+                .icon(() -> icon)
+                .build();
+
+        groups.put(name, new Group(itemGroup, icon, items));
+
+        reloadGroups(group.getItemGroup().getIndex());
+        saveFile();
+        sendFeedback("commands.citemgroup.changeIcon.success", name, old.getItem().getName(), icon.getItem().getName());
+        return 0;
+    }
+
+    private static int renameGroup(ServerCommandSource source, String name, String _new) throws CommandSyntaxException {
+        if (!groups.containsKey(name)) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        Group group = groups.remove(name);
+        ItemGroup itemGroup = group.getItemGroup();
+        ListTag items = group.getItems();
+        ItemStack icon = itemGroup.getIcon();
+        try {
+            itemGroup = FabricItemGroupBuilder.create(
+                    new Identifier("clientcommands", _new))
+                    .icon(() -> icon)
+                    .build();
+
+            groups.put(_new, new Group(itemGroup, icon, items));
+        } catch (InvalidIdentifierException e) {
+            throw ILLEGAL_CHARACTER_EXCEPTION.create(_new);
+        }
+
+        reloadGroups(group.getItemGroup().getIndex());
+        saveFile();
+        sendFeedback("commands.citemgroup.renameGroup.success", name, _new);
+        return 0;
+    }
+
+    private static void saveFile() throws CommandSyntaxException {
+        try {
+            CompoundTag rootTag = new CompoundTag();
+            CompoundTag compoundTag = new CompoundTag();
+            groups.forEach((key, value) -> {
+                CompoundTag group = new CompoundTag();
+                group.put("icon", value.getIcon().toTag(new CompoundTag()));
+                group.put("items", value.getItems());
+                compoundTag.put(key, group);
+            });
+            rootTag.putInt("DataVersion", SharedConstants.getGameVersion().getWorldVersion());
+            rootTag.put("Groups", compoundTag);
+            File newFile = File.createTempFile("groups", ".dat", configPath.toFile());
+            NbtIo.write(rootTag, newFile);
+            File backupFile = new File(configPath.toFile(), "groups.dat_old");
+            File currentFile = new File(configPath.toFile(), "groups.dat");
+            Util.backupAndReplace(currentFile, newFile, backupFile);
+        } catch (IOException e) {
+            throw SAVE_FAILED_EXCEPTION.create();
```

Comment:
Printing the stack trace here might be useful to help people figure out what's going wrong